src/wasm/wasm-binary.cpp

```diff
@@ -945,6 +945,15 @@ void WasmBinaryBuilder::read() {
 
     auto oldPos = pos;
 
+    // almost no sections can appear more than once
+    if (seenSections.count(BinaryConsts::Section(sectionCode)) > 0) {
+      if (sectionCode != BinaryConsts::Section::User && sectionCode != BinaryConsts::Section::Code) {
+        throw ParseException("section seen more than once: " + std::to_string(sectionCode));
+      }
+    } else {
+      seenSections.insert(BinaryConsts::Section(sectionCode));
```

Comment:
Could just directly call `insert` and check the second element of its return value to see if there was a duplicate. Not as big of a performance difference obviously as the other PR but maybe still worth it.